docs/features/features.md

```diff
@@ -0,0 +1,207 @@
+# NGINX Agent Features
+
+- [Introduction](#introduction)
+- [Feature Flags Overview](#feature-flags-overview)
+- [Conflicting Combinations](#conflicting-combinations)
+- [Cobra CLI Parameters](#cobra-cli-parameters)
+- [Environment Variables](#environment-variables)
+- [Configuration File](#configuration-file)
+- [Dynamic Updates via gRPC](#dynamic-updates-via-grpc)
+- [Internal State Management](#internal-state-management)
+- [Code Path Management](#code-path-management)
+- [Dynamic Feature Toggling](#dynamic-feature-toggling)
+- [Security Considerations](#security-considerations)
+- [Conclusion](#conclusion)
+
+## Introduction
+
+This document outlines the design of a feature flag management system
+for the NGINX Agent. The system enables dynamic toggling of features
+based on configurations provided via multiple sources, including CLI
+parameters, environment variables, configuration files, and via protobuf
+gRPC messages. Feature flags control the execution paths within the
+application, allowing for flexible and controlled feature management.
+
+## Feature Flags Overview
+
+The design goal of this document is to capture fine grained features in
+NGINX Agent v3.
+
+The ultimate goal of this design is to delegate fine-grained control of
+high-level feature sets to the configuration.
+
+The system manages the following feature flags:
+
+| Feature Flag  | Sub Category 1    | Description                                                                                                  | Default            |
+|---------------|-------------------|--------------------------------------------------------------------------------------------------------------|--------------------|
+| configuration |                   | Full read/write management of configurations toggled by DataPlaneConfig ConfigMode                            | On                 |
+| certificates  |                   | Inclusion of public keys and other certificates in the configurations toggled by DataPlaneConfig CertMode      | Off                |
+| connection    |                   | Sends an initial connection message reporting instance information on presence of Command ServerConfig Host and Port | On            |
+| file-watcher  |                   | Monitoring of file changes in allowed directories; may trigger DP => MP sync of files configured by Watchers   | On                 |
```

Comment:
Should there be a key somewhere for the acronyms here or the full words be used instead ?